src/key/schnorr.ts

```diff
@@ -25,9 +25,16 @@ export interface SchnorrSignature {
 /**
  * @hidden
  */
-function schnorrHash(r: BN, msg: BN): string {
-    const rString = r.toBuffer("be", 32).toString("binary");
-    const mString = msg.toBuffer("be", 32).toString("binary");
+export function schnorrHash(r: BN, msg: BN): string {
+    const rString = r
+        .toBuffer("be", 32)
+        .toString("binary")
+        .padStart(32, "\x00");
+    const mString = msg
+        .toBuffer("be", 32)
+        .toString("binary")
+        .padStart(32, "\x00");
```

Comment:
Please add a simple comment about why we need padding.

Reply:
I added some explanation